src/bin/ion/commands/beta/generate/templates/java/class.templ

```diff
@@ -16,4 +21,110 @@ public final class {{ target_kind_name }} {
         return this.{{ field.name | camel }};
     }
     {% endfor %}
+
+    public static {{ target_kind_name }} readFrom(IonReader reader) {
```

Comment:
(🗺️ PR tour) Added `readFrom` API in the class template for reading an Ion value into the given data model.